xdr/transaction_result.go

```diff
@@ -30,3 +32,27 @@ func (r TransactionResultPair) OperationResults() ([]OperationResult, bool) {
 func (r TransactionResultPair) InnerHash() Hash {
 	return r.Result.Result.MustInnerResultPair().TransactionHash
 }
+
+// ExtractBalanceId will parse the operation result at `opIndex` within the
+// given `txResult`.
+//
+// If the specified operation index does not point to a successful
+// `CreateClaimableBalance` operation result, this function panics.
+func (r TransactionResult) ExtractBalanceID(opIndex int) (string, error) {
```

Comment:
LGTM! One suggestion I have is to return `ClaimableBalanceId` and create a helper method that returns hex (`ExtractBalanceIDHex`). This would be consistent with other methods (like `Marshal` -> `MarshalHex`).
```suggestion
func (r TransactionResult) ExtractBalanceID(opIndex int) (ClaimableBalanceId, error) {
```

Reply:
Hmm, though I'm having trouble envisioning a use case for the raw XDR structure, I've added it as suggested for consistency in 41537b4927d3cf510a2708002da3a1f065edd552!